template.go

```diff
@@ -207,6 +209,32 @@ const (
 				}
 			{{end}}
 
+			func ({{$m}} *m{{$mock}}{{$method.Name}}{{(paramsRef)}}) Times(n uint64) *m{{$mock}}{{$method.Name}}{{(paramsRef)}} {
+				if n == 0 {
+					{{$m}}.mock.t.Fatalf("Times of {{$mock}}.{{$method.Name}} mock can not be zero")
+				}
+				mm_atomic.StoreUint64(&{{$m}}.expectedInvocations, n)
+				return {{$m}}
+			}
+
+			func ({{$m}} *m{{$mock}}{{$method.Name}}{{(paramsRef)}}) invocationsDone() bool {
+				if len({{$m}}.expectations) == 0 && {{$m}}.defaultExpectation == nil && {{$m}}.mock.func{{$method.Name}} == nil {
+					// does not need to check invocations if no expectations, defaultExpectation or func{{$method.Name}} set
+					return true
+				}
+
+				// if expectations were set we check total invocations
+				// if default expectation was set then invocations count should be greater than zero
+				// if func was set then invocations count should be greater than zero
+				totalInvocations := mm_atomic.LoadUint64(&{{$m}}.mock.after{{$method.Name}}Counter)
+				expectedInvocations := mm_atomic.LoadUint64(&{{$m}}.expectedInvocations)
+				if totalInvocations < 1 || expectedInvocations != 0 && expectedInvocations != totalInvocations {
```

Comment:
This should be enough here, I think:
```go
return expectedInvocations == 0 || expectedInvocations == totalInvocations
```

Reply:
Actually we have to check that totalInvocations > 0 to remain old logic. Otherwise if we didnt set Times (expectedInvocations is 0) and no calls was done (totalInvocations is 0) we wont get `Expected call to ` error. 
I ve added test `TestContextAccepterMock_ExpectedCall` on this behaviour

Reply:
Oh, I see, then it should be something like this:
```go
return totalInvocations > 0 && (expectedInvocations == 0 || expectedInvocations == totalInvocations)
```